### ISSUE TITLE
Static rules generate empty CSS rules in dynamic style sheet when a function rule is used

### ISSUE THREAD

**Issue by wouter-willems** [cssinjs/react-jss#267]
Hi,
whenever i am using dynamic properties, like:

```
wrapper: (props) => ({
    padding: 40,
    background: theme.background,
    textAlign: "left"
  }),
```

It causes an extra class to be added, which is empty and does nothing. You can see an example in this playground for the wrapper:

https://codesandbox.io/s/j3l06yyqpw

the result: `class="Comp-wrapper-0-1-10 Comp-wrapper-0-1-7"`

In that case, `Comp-wrapper-0-1-7` does nothing. You can see that even all children now have double classes.
This also does not work for the newest version, 9.8.7.

**Comment by kof** [comment]
I see this: and I see that empty CSS rules are generated in the dynamic style instance: title and link classes should not be there, since they are fully static and are already rendered in the previous style.

<img width="340" alt="screen shot 2018-07-03 at 12 56 12 pm" src="https://user-images.githubusercontent.com/52824/42215992-8b450c18-7ec0-11e8-9fd1-6f9f740ea6fb.png">

**Comment by wouter-willems** [comment]
What about the duplicate 'wrapper' class that is injected onto the dom element?

**Comment by kof** [comment]
Oh yes thats a bug too, there is no even such CSS rule generated.

**Comment by kof** [comment]
It is all related to function rules, I recommend to not use them for now and use function values only for dynamic values where really needed, since they don't support full syntax either, here is the issue https://github.com/cssinjs/jss/issues/682

**Comment by wouter-willems** [comment]
Great, thank you kof! Looking forward to the fixes =)

**Comment by HenriBeck** [comment]
For the rules in the dynamic sheet we could just remove any rule which is an emtpy object.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kof** [comment]
I think there is a bug in the dynamic rules extraction, there should be no rules which are completely static in the dynamic sheet.

**Comment by HenriBeck** [comment]
Yeah, so it's an issue of getDynamicStyles function in jss. But we currently use that "bug" for the classes prop. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by HenriBeck** [comment]
The empty style rules are caused because we are depending here on `jss-compose` when we have styles with dynamic values. I think this should be removed.

**Comment by kof** [comment]
ah, there should be a way to compose without relying on jss-compose here, we can compose those classes within react-jss when we build classes map

**Comment by kof** [comment]
published 8.6.1

**Comment by kof** [comment]
good work @HenriBeck
